metals/src/main/scala/scala/meta/internal/metals/WorkspaceSearchVisitor.scala

```diff
@@ -86,10 +86,14 @@ class WorkspaceSearchVisitor(
   ): Option[SymbolDefinition] = {
     val nme = Classfile.name(filename)
     val tpe = Symbol(Symbols.Global(pkg, Descriptor.Type(nme)))
-    index.definition(tpe).orElse {
+
+    val forTpe = index.definitions(tpe)
+    val defs = if (forTpe.isEmpty) {
       val term = Symbol(Symbols.Global(pkg, Descriptor.Term(nme)))
-      index.definition(term)
-    }
+      index.definitions(term)
+    } else forTpe
+
+    defs.sortBy(_.path.toURI.toString).headOption
```

Comment:
Just to make sure I understand this. In your example with:

```
io:netty.netty-all:4.0.43.Final - io.netty.util.collection.IntObjectMap#Entry
io.netty.netty-common:4.1.17.Final - io.netty.util.collection.IntObjectMap#PrimitiveEntry
```
This is to ensure that we are always getting the same one back right? But aren't these technically two different symbols? How come `index.definition(tpe)` was returning a random one if they are unique? I'm not super familiar with the GlobalSymbolIndex so I may just totally be missing something.

Reply:
They are different but here `index.definitions` is performed only to find the location of a top-level class.

In this test  classpath search found these two class files wich has name that fit query:
  - `io/netty/util/colletion/IntObjectMap$Entry.class`
  - `io/netty/util/colletion/IntObjectMap$PrimitiveEntry.class`

Then it calls `index.definition` to find toplevel symbols `io.netty.util.collection.IntObjectMap` to perform actual matching against semantic defitions extracted from source file.

Reply:
Ahh ok, thanks for explaining 👍🏼